docs/tutorials/intro.md

```diff
@@ -108,43 +113,52 @@ fn main() {
+The way `ICU4X` plugs into that dataset is one of its novelties, aiming at making the data management more flexible and enabling better integration in asynchronous environments.
 
-In result, compared to most internationalization solutions, working with `ICU4X` and data is a bit more explicit. `ICU4X` provides a trait called `DataProvider` and a number of concrete APIs that implement that trait for different scenarios.
+In result, compared to most internationalization solutions, working with data in `ICU4X` is a bit more explicit. `ICU4X` provides a trait called `DataProvider` (as well as `BufferProvider` and `AnyProvider`) and a number of concrete APIs that implement these traits for different scenarios.
 Users are also free to design their own providers that best fit into their ecosystem requirements.
 
+`BufferProvider` and `AnyProvider` abstract over different ways the data may be loaded: `BufferProvider` abstracts over data providers that deserialize data, whereas `AnyProvider` abstracts over data providers that can provide concrete Rust objects, like data providers that use [`databake`](https://docs.rs/databake).
+
 In this tutorial we are going to use ICU4X's "test" data provider and then move on to a synchronous file-system data provider which uses ICU4X format JSON resource files.
 
 ## Test data
 
-ICU4X's repository comes with pre-generated test data that covers all of its keys for a select set of locales. For production use it is recommended one use the steps in [Generating Data](#generating-data) to generate custom data, but for the purposes of trying stuff out, it is sufficient to use the data providers exported by `icu_testdata`.
+ICU4X's repository comes with pre-generated test data that covers all of its keys for a select set of locales. For production use it is recommended one use the steps in [Generating Data](#generating-data) to generate custom data, but for the purposes of trying stuff out, it is sufficient to use the data providers exported by `icu_testdata`. `icu_testdata::any()` will produce an `AnyProvider`, and `icu_testdata::buffer()` will produce a `BufferProvider`.
+
+# 6. Using an ICU4X component
 
+We're going to try writing an app that uses the `icu::datetime` component to format a date and time in a Japanese locale.
 
-## Using test data
+First, we need to register our choice of the provider in `~/projects/icu_tutorial/myapp/Cargo.toml`, which can be done by running:
 
-First, we need to register our choice of the provider in `~/projects/icu/myapp/Cargo.toml`:
+```sh
+$ cargo add icu_testdata
+```
+
+which should generate:
 
 ```
 [dependencies]
 icu = "1.0"
 icu_testdata = "1.0"
 ```
 
-and then we can use it in our code:
+We can then use it in our code:
 
 ```rust
 fn main() {
-    let _provider = icu_testdata::unstable();
+    let _provider = icu_testdata::any();
```

Comment:
This is weird, we're replacing all the "progress" we've made. Can we not iterate on the code we already have, i.e. keeping the locale declaration and print statement, just adding a date to it?

Reply:
I think the examples get bigger if we do that, and it's really useful to be able to _focus_ on the important part.

Reply:
I think that's fair, i also think that might need more holistic work

But I do think it's important to show snippets _before_ iterating. The snippets can be more clearly marked as such though.